[2024-07-17T01:23:49.127+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-17T01:23:49.175+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_ticker.etl_usd_mxn scheduled__2024-07-15T08:30:00+00:00 [queued]>
[2024-07-17T01:23:49.195+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_ticker.etl_usd_mxn scheduled__2024-07-15T08:30:00+00:00 [queued]>
[2024-07-17T01:23:49.196+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2024-07-17T01:23:49.216+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): etl_usd_mxn> on 2024-07-15 08:30:00+00:00
[2024-07-17T01:23:49.238+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=30816) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-07-17T01:23:49.240+0000] {standard_task_runner.py:63} INFO - Started process 30833 to run task
[2024-07-17T01:23:49.236+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'etl_ticker', 'etl_usd_mxn', 'scheduled__2024-07-15T08:30:00+00:00', '--job-id', '758', '--raw', '--subdir', 'DAGS_FOLDER/bid_ask_dag.py', '--cfg-path', '/tmp/tmp3ykfu4qp']
[2024-07-17T01:23:49.252+0000] {standard_task_runner.py:91} INFO - Job 758: Subtask etl_usd_mxn
[2024-07-17T01:23:49.390+0000] {task_command.py:426} INFO - Running <TaskInstance: etl_ticker.etl_usd_mxn scheduled__2024-07-15T08:30:00+00:00 [running]> on host 6f712de23803
[2024-07-17T01:23:49.546+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='etl_ticker' AIRFLOW_CTX_TASK_ID='etl_usd_mxn' AIRFLOW_CTX_EXECUTION_DATE='2024-07-15T08:30:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-07-15T08:30:00+00:00'
[2024-07-17T01:23:49.553+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-17T01:23:49.589+0000] {logging_mixin.py:188} WARNING -   0%|          | 0/600 [00:00<?, ?it/s]
[2024-07-17T01:23:51.032+0000] {bid_ask_dag.py:87} INFO - Threshold! {'orderbook_timestamp': '2024-07-17T01:23:49+00:00', 'book': 'usd_mxn', 'bid': 17.62, 'ask': 17.64, 'spread': 0.11337868480725381}
[2024-07-17T01:23:51.037+0000] {logging_mixin.py:188} WARNING -   0%|          | 0/600 [00:01<?, ?it/s]
[2024-07-17T01:23:51.040+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-17T01:23:51.043+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/bid_ask_dag.py", line 88, in etl_spread
    SlackWebhookHook(webhook_token=WEBHOOK_URL, message=f"Alert! bid-ask has appeared {record}").execute()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: SlackWebhookHook.__init__() missing 1 required keyword-only argument: 'slack_webhook_conn_id'
[2024-07-17T01:23:51.058+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=etl_ticker, task_id=etl_usd_mxn, run_id=scheduled__2024-07-15T08:30:00+00:00, execution_date=20240715T083000, start_date=20240717T012349, end_date=20240717T012351
[2024-07-17T01:23:51.072+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 758 for task etl_usd_mxn (SlackWebhookHook.__init__() missing 1 required keyword-only argument: 'slack_webhook_conn_id'; 30833)
[2024-07-17T01:23:51.104+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-07-17T01:23:51.120+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-17T01:23:51.121+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
